pkg/utils/secrets/manager/generate.go

```diff
@@ -148,6 +155,10 @@ func (m *manager) keepExistingSecretsIfNeeded(ctx context.Context, configName st
 			return nil, err
 		}
 
+		if configName != "kube-apiserver-basic-auth" {
```

Comment:
I would be explicit here:
```suggestion
		if configName == "observability-ingress" ||  configName ==  "observability-ingress-users" {
```
a bit more straight-forward to read and guards against future bugs in case you add other old secrets which should be kept (which might not want BasicAuthFormatNormal).